crates/sui-storage/src/lib.rs

```diff
@@ -44,3 +45,14 @@ pub fn default_db_options(
 
     (options, point_lookup)
 }
+
+// Used to exec futures that send data to/from other threads. In the simulator, this effectively
+// becomes a blocking call, which removes the non-determinism that would otherwise be caused by the
+// timing of the reply from the other thread.
+pub(crate) async fn exec_client_future<F: Future>(fut: F) -> <F as Future>::Output {
```

Comment:
I think I understand why you need this function, but I don't understand choice of the name to be honest :)

Maybe something like `sim_blocking_call` or `sim_no_preempt` would be better name? It would give intuition that 
(a) this is done as a call specifically to make it work in simulator
(b) what it helps specifically with is communicating with threads that are outside of simulator scope

Reply:
Good suggestion.